queries/php/highlights.scm

```diff
@@ -55,8 +55,9 @@
 ((name) @constant
  (#vim-match? @constant "^_?[A-Z][A-Z\d_]+$"))
 
-((name) @constructor
- (#lua-match? @constructor "^[A-Z]"))
+(method_declaration
+    name: (name) @constructor
+    (#eq? @constructor "__construct"))
```

Comment:
I guess the query is trying to highlight the case `new BaseClass()` in 
```
class BaseClass {
    function __construct() {
        print "Im BaseClass-Konstruktor\n";
    }
}

class SubClass extends BaseClass {
    function __construct() {
        parent::__construct();
        print "Im SubClass-Konstruktor\n";
    }
}

class OtherSubClass extends BaseClass {
    // erbt den Konstruktor von BaseClass
}

// Im BaseClass-Konstruktor
$obj = new BaseClass();
```
you could swap the queries so that constant has higher priority or just highlight this case in combination with the `new` keyword. But I'd be also fine with this solution.

Reply:
@theHamsta The object creation could be matched by this query:
```
(object_creation_expression (qualified_name) @constructor)
```

Reply:
@theHamsta 
![Screenshot from 2022-01-15 08-18-24](https://user-images.githubusercontent.com/57968042/149610108-478e9d04-b61c-4a2a-aa53-38b44fe4c723.png)
![Screenshot from 2022-01-15 08-18-40](https://user-images.githubusercontent.com/57968042/149610109-2503d188-4b1e-4dc4-adb5-8a7a75bb1cfe.png)
Do we need whole creation expression or just the name?

@mikehaertl
![Screenshot from 2022-01-15 08-18-52](https://user-images.githubusercontent.com/57968042/149610142-e08f9051-618c-4d2c-9bc1-404c8ad24aed.png)
(qualified_name) doesn't match. Do you have different results?

Reply:
@infectos It can be `name`  (class without path) or `qualified_name` (fully qualified name like `\some\TestClass`). An alternation expression should work:

```
(object_creation_expression
  [(name) (qualified_name)] @constructor)
```

Reply:
If this is getting to complicated we can move the highlight of object creation as constructor to another PR or omit it entirely. I'd be fine with merging when you are.

Reply:
@theHamsta I think you can merge it. I'm not even sure if it's a good idea to mark object creation as `TSConstruct`.